library/src/main/java/com/kizitonwose/calendarview/CalendarView.kt

```diff
@@ -619,6 +622,49 @@ open class CalendarView : RecyclerView {
         )
     }
 
+    /**
+     * Update the CalendarView's month range. This method does the minimal updates to the
+     * underlying RecyclerView resulting in a stable view when the changes don't affect the visible
+     * month(s).
+     */
+    fun updateRange(startMonth: YearMonth, endMonth: YearMonth) {
+        val oldStartMonth = this.startMonth!!
+        val oldEndMonth = this.endMonth!!
+        this.startMonth = startMonth
+        this.endMonth = endMonth
+
+        calendarAdapter.monthConfig =
+            MonthConfig(
+                outDateStyle, inDateStyle, maxRowCount,
+                startMonth, endMonth, firstDayOfWeek ?: return,
+                hasBoundaries
+            )
+        DiffUtil.calculateDiff(
+            MonthRangeDiffCallback(oldStartMonth, oldEndMonth, startMonth, endMonth), false)
+            .dispatchUpdatesTo(calendarAdapter)
+    }
+
+    private class MonthRangeDiffCallback(
```

Comment:
This diffing logic will not work if the calendar's `maxRowCount` value is not set to 6. For example, in a week mode calendar(`maxRowCount` is 1), a `YearMonth` will be represented over about 4 or 5 indices so getting the list count by counting the number of months between the start and end months will give an inaccurate result.

A better approach would be to use the old and the newly created `MonthConfig` for diffing, There's a `months` property in it which gives the accurate size.

Reply:
Good point, done.